library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js

```diff
@@ -28,9 +29,20 @@ $.fn.folderListing = function(options) {
 
   var urlPrefix = '/direct/content/resources/';
 
-  var getHtmlResults = function(json, $div) {
+  var getHtmlResults = function(json, $div, root) {
     var html = $('<ul/>').addClass('jqueryFileTree').hide();
-
+    var urlRoot = json[0].resourceId.trim();
+    if(settings.displayRootDirectory && urlRoot === root ){
```

Comment:
Better to extract this out as a variable, especially as the same test is done in 2 places and they are a long way apart in the code.